htdocs/product/list.php

```diff
@@ -594,20 +594,22 @@
 //if (GETPOST("toolowstock")) $sql.= " HAVING SUM(s.reel) < p.seuil_stock_alerte";    // Not used yet
 
 $nbtotalofrecords = '';
-if (empty($conf->global->MAIN_DISABLE_FULL_SCANLIST)) {
-	/* $result = $db->query($sql);
-	$nbtotalofrecords = $db->num_rows($result);
-	*/
-	/* The fast and low memory method to get and count full list converts the sql into a sql count */
-	$sqlforcount = preg_replace('/^SELECT[a-zA-Z0-9\._\s\(\),=<>\:\-\']+\sFROM/Ui', 'SELECT COUNT(*) as nbtotalofrecords FROM', $sql);
-	$sqlforcount = preg_replace('/'.preg_quote($linktopfp, '/').'/', '', $sqlforcount);
-	$sqlforcount = preg_replace('/GROUP BY .*$/', '', $sqlforcount);
-	$resql = $db->query($sqlforcount);
-	if ($resql) {
-		$objforcount = $db->fetch_object($resql);
-		$nbtotalofrecords = $objforcount->nbtotalofrecords;
+if (!getDolGlobalInt('MAIN_DISABLE_FULL_SCANLIST')) {
+	if (getDolGlobalInt('MAIN_FEATURES_LEVEL') < 2) {
+		$result = $db->query($sql);
+		$nbtotalofrecords = $db->num_rows($result);
```

Comment:
This method must be abandonned. It download all the database into php memory.
Can you describe the trouble you have instead insto a ticket, and put the sql you get after this if ?

Reply:
Thanks.
For the subselect in hook I have this SQL issue:
`SELECT COUNT(*) as nbtotalofrecords FROM llx_product_stock WHERE fk_product = p.rowid AND fk_entrepot = 20) as warehouse_stock FROM llx_product as p LEFT JOIN llx_product_extrafields as ef on (p.rowid = ef.fk_object) LEFT JOIN llx_product_lang as pl ON pl.fk_product = p.rowid AND pl.lang = 'fr_FR' LEFT JOIN llx_c_units cu ON cu.rowid = p.fk_unit WHERE p.entity IN (1) AND p.fk_product_type <> 1`

Part from hook which is not removed from select = `FROM llx_product_stock WHERE fk_product = p.rowid AND fk_entrepot = 20) as warehouse_stock`